File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EInverseSurfaceLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EInverseSurfaceLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl7
            AppTheme.STANDARD -> R.color.cstl7
            AppTheme.BLUE -> R.color.cbl7
            AppTheme.GREEN -> R.color.cgl7
            AppTheme.RED -> R.color.crl7
            AppTheme.NOTHING -> R.color.cnol7
        }
    }
}

File: lib/layer2_core/light/e_inverse_surface_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EInverseSurfaceLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl7;
      case 'standard': return cstl7;
      case 'blue': return cbl7;
      case 'green': return cgl7;
      case 'red': return crl7;
      case 'nothing': return cnol7;
      default: return cstl7;
    }
  }
}
